Makefile

```diff
@@ -306,6 +306,10 @@ lint: check-makefiles
 	# Ensure all errors are report as APIError
 	faillint -paths "github.com/weaveworks/common/httpgrpc.{Errorf}=github.com/grafana/mimir/pkg/api/error.Newf" ./pkg/frontend/querymiddleware/...
 
+	# Do not use grpc/status.FromError() because doesn't support wrapped errors.
```

Comment:
This is actually not true: 
- `grpc/status.FromError()` support wrapped errors (it uses `errors.As()` internally)
- `gogo/status.FromError()` doesn't support wrapped errors (it doesn't use `errors.As()` internally)

So, I'd suggest to decommission both `grpc/status.FromError()` and `gogo/status.FromError()` and use `grpcutil.ErrorToStatus()` instead.

Reply:
I think I was completely drunk when I worked on this PR 🤦

Reply:
@duricanikolic I should have fix it.